core/base/src/TSystem.cxx

```diff
@@ -2922,6 +2922,14 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
       incPath.Append(fromConfig);
    }
    incPath.ReplaceAll(" -I",":");       // of form :dir1 :dir2:dir3
+   auto posISysRoot = incPath.Index(" -isysroot \"");
+   if (posISysRoot != kNPOS) {
+      auto posISysRootEnd = incPath.Index('"', posISysRoot + 12);
+      if (posISysRootEnd != kNPOS) {
+         incPath.Insert(posISysRootEnd - 1, "/usr/include/");
```

Comment:
Why hard-coding `/usr/include` here. This will make the effort of cross-compiling harder.

Reply:
This picks up sysroot + "/usr/include" - but indeed it should pick up "all -I prefixed with sysroot". I'll add a note or implement it.